lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2214,7 +2229,33 @@ LLVMToSPIRVBase::transValueWithoutDecoration(Value *V, SPIRVBasicBlock *BB,
     unsigned ArgNo = Arg->getArgNo();
     SPIRVFunction *BF = BB->getParent();
     // assert(BF->existArgument(ArgNo));
-    return mapValue(V, BF->getArgument(ArgNo));
+    auto *SPVArg = BF->getArgument(ArgNo);
+
+    if (BM->isAllowedToUseExtension(ExtensionID::SPV_KHR_untyped_pointers) &&
+        SPVArg->getType()->isTypePointer() &&
+        !SPVArg->getType()->isTypeUntypedPointerKHR()) {
+      // When SPV_KHR_untyped_pointers extension is enabled, bitcast typed
+      // pointer function arguments to untyped pointers for further usage in the
+      // untyped pointers paradigm.
+      // Don't do this if the argument is further used in function calls as it
+      // breaks SPIR-V validity.
+      for (auto *U : V->users()) {
+        if (isa<CallInst>(U) || isa<InvokeInst>(U) || isa<PHINode>(U))
```

Comment:
Why phi instruction is also guarded?

Reply:
Yep, this can be safely removed, thanks for noticing!